java/flight/src/main/java/org/apache/arrow/flight/FlightService.java

```diff
@@ -221,6 +221,19 @@ public void getFlightInfo(Flight.FlightDescriptor request, StreamObserver<Flight
     }
   }
 
+  @Override
+  public void getFlightSchema(Flight.FlightDescriptor request, StreamObserver<Flight.SchemaResult> responseObserver) {
+    try {
+      SchemaResult result = producer
+              .getFlightSchema(makeContext((ServerCallStreamObserver<?>) responseObserver),
+                      new FlightDescriptor(request));
+      responseObserver.onNext(result.toProtocol());
+      responseObserver.onCompleted();
+    } catch (Exception ex) {
+      responseObserver.onError(ex);
```

Comment:
This should use `StatusUtils.toGrpcException`.

Reply:
Fixed